### PR TITLE
Discussion-only PR for Volkswagen FPv2

### PR DESCRIPTION
**Pull request by jyoung8607**
Do not merge, discussion only. CI expected to fail.

This is an example of minimally viable FPv2 for Volkswagen MQB. It adds support for different rx_addr offsets to the UDS and ISO-TP code in a way that tries to avoid affecting queries for other cars. Identification is done by querying the steering characteristic parameters from the EPS rack. A version of this does work on my development vehicle.

Further discussion: https://github.com/commaai/openpilot/issues/1238#issuecomment-620142101

**TL;DR:** I am not in favor of the steering-rack parameter ONLY approach to identification, but I'll implement it if that's what Comma wants. I still strongly advocate for using the VIN, backed up by UDS validation of versions, as the most correct and future-proof solution. It could be done in a way that only affects car port(s) that want to use the VIN as part of the identity mix.

Thoughts requested from @gregjhogan and @pd0wm based on our DM discussions.

### CODE REVIEW

**Comment by jyoung8607** [comment]
This would also include a change to the Panda UDS library at this location:

https://github.com/commaai/panda/blob/ba59ada0e95df66bcca28d858fbf2daf6a2a7e40/python/uds.py#L471

To this:

```python
def get_rx_addr_for_tx_addr(tx_addr, rx_offset=0x8):
  if tx_addr in FUNCTIONAL_ADDRS:
    return None

  if tx_addr < 0xFFF8:
    # Standard 11 bit response addr (add 8), unless overridden
    # Some manufacturers use different offsets
    return tx_addr + rx_offset
```

Just didn't want to carry around a second Panda PR for this discussion.

**Comment by pd0wm** [comment]
As discussed:

All MQB cars have the same ecu fw versions, and they speak the same API. For openpilot there will be only one car. CAR.VW_MQB. The only place we need to distinguish between car models is in the get_params to get to correct wheelbase and tuning params. For now we'll use VIN there, if in the future that doesn't work we can very easily switch to something else (e.g. the EPS has an dataflash id). By solving it this way the FW query is the same as all other cars, and the VIN code is very contained.

**Comment by jyoung8607** [comment]
Withdrawing for now, per our discussion. Will take another run at this in late May/early June.
